### ISSUE TITLE
IDE0058 does not fail the build when it appears as top level statement

### ISSUE THREAD

**Issue by WelliSolutions** [dotnet/roslyn-analyzers#7408]
### Analyzer

**Diagnostic ID**: [IDE0058](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0058): `Remove unnecessary expression value`

### Analyzer source

**SDK**: Built-in CA analyzers in .NET 8 SDK 

**Version**: I don't know exactly, whatever comes with Visual Studio 2022 (Version 17.11.2)

### Describe the bug

```ini
dotnet_diagnostic.IDE0058.severity = error
```

does not make the build fail for the code

```csharp
System.Convert.ToInt32("35");
```

Note: this is all code. Top level statements.

### Steps To Reproduce

1. Create an empty .NET 8 Console application. Then replace all code with the C# source above.
2. Create an empty `.editorconfig` file from the Visual Studio template, then add the IDE0058 rule as shown above.
3. Rebuild.

### Expected behavior

IDE0058 is shown as squiggly lines in the editor. 

IDE0058 is reported as an error during build.

### Actual behavior

IDE0058 is shown as squiggly lines in the editor. 

The build passes.

The build fails, when the same line of code is inside a `Program` class, `Main()` method.

## Additional context

Full description of everything is on [Stack Overflow](https://stackoverflow.com/questions/78951936/how-to-enable-code-analysis-ide0058-as-a-build-error)
